rectools/models/bert4rec.py

```diff
@@ -77,16 +79,26 @@ def _collate_fn_train(
             y[i, -len(ses) :] = target  # ses: [session_len] -> y[i]: [session_max_len + 1]
             yw[i, -len(ses) :] = ses_weights  # ses_weights: [session_len] -> yw[i]: [session_max_len + 1]
 
-        return torch.LongTensor(x), torch.LongTensor(y), torch.FloatTensor(yw)
-
-    def _collate_fn_recommend(self, batch: List[Tuple[List[int], List[float]]]) -> torch.LongTensor:
+        # TODO: we are sampling negatives for paddings. Let's think about it later
+        batch_dict = dict({"x": torch.LongTensor(x), "y": torch.LongTensor(y), "yw": torch.FloatTensor(yw)})
+        # TODO: we are sampling negatives for paddings. Let's think about it later
+        if self.n_negatives is not None:
+            negatives = torch.randint(
+                low=self.n_item_extra_tokens,
+                high=self.item_id_map.size,
+                size=(batch_size, self.session_max_len, self.n_negatives),
+            ).long()  # [batch_size, session_max_len, n_negatives]
+            batch_dict["negatives"] = torch.LongTensor(negatives)
```

Comment:
You have specified Long dtype for this tensor 3 times in a row :)
First `torch.randint` has default dtype `torch.int64`
Then you do `long()`
And finally `torch.LongTensor`

Reply:
fixed